artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/AMQPBrokerConnection.java

```diff
@@ -0,0 +1,556 @@
+            AMQPMirrorControllerAggregation remoteAggregation = new AMQPMirrorControllerAggregation();
+            remoteAggregation.addPartition((AMQPMirrorControllerSource) currentMirrorController);
+            currentMirrorController = remoteAggregation;
+            server.installMirrorController(remoteAggregation);
+         }
+         ((AMQPMirrorControllerAggregation) currentMirrorController).addPartition(newPartition);
+      }
+
+      return snfReplicaQueueBinding;
+   }
+
+   private void connectReceiver(ActiveMQProtonRemotingConnection protonRemotingConnection,
+                                Session session,
+                                AMQPSessionContext sessionContext,
+                                Queue queue,
+                                Symbol... capabilities) {
+      if (logger.isDebugEnabled()) {
+         logger.debug("Connecting inbound for " + queue);
+      }
+
+      if (session == null) {
+         logger.debug("session is null");
+         return;
+      }
+
+      protonRemotingConnection.getAmqpConnection().runLater(() -> {
+
+         if (receivers.contains(queue)) {
+            logger.debug("Receiver for queue " + queue + " already exists, just giving up");
+            return;
+         }
+         receivers.add(queue);
+         Receiver receiver = session.receiver(queue.getName().toString() + UUIDGenerator.getInstance().generateStringUUID());
+         receiver.setSenderSettleMode(SenderSettleMode.UNSETTLED);
+         receiver.setReceiverSettleMode(ReceiverSettleMode.FIRST);
+         Target target = new Target();
+         target.setAddress(queue.getAddress().toString());
+         receiver.setTarget(target);
+
+         Source source = new Source();
+         source.setAddress(queue.getAddress().toString());
+         receiver.setSource(source);
+
+         if (capabilities != null) {
+            source.setCapabilities(capabilities);
+         }
+
+         receiver.open();
+         protonRemotingConnection.getAmqpConnection().flush();
+         try {
+            sessionContext.addReceiver(receiver);
+         } catch (Exception e) {
+            error(e);
+         }
+      });
+   }
+
+   private void connectSender(Queue queue,
+                              String targetName,
+                              Symbol... capabilities) {
+      if (logger.isDebugEnabled()) {
+         logger.debug("Connecting outbound for " + queue);
+      }
+
+
+      if (session == null) {
+         logger.debug("Session is null");
+         return;
+      }
+
+      protonRemotingConnection.getAmqpConnection().runLater(() -> {
+         try {
+            if (senders.contains(queue)) {
+               logger.debug("Sender for queue " + queue + " already exists, just giving up");
+               return;
+            }
+            senders.add(queue);
+            Sender sender = session.sender(targetName + UUIDGenerator.getInstance().generateStringUUID());
```

Comment:
Adding a ":" between the name and generated bit would aid readability when debugging protocol trace.